Calendars/Calendars.Plugin.Tizen/CalendarsImplementation.cs

```diff
@@ -0,0 +1,490 @@
+				sTime = calRecord.Get<CalendarTime>(Event.Start);
+				time = new CalendarTime(sTime.UtcTime.AddMinutes(reminder?.TimeBefore.TotalMinutes ?? 15).Ticks);
+				calAlarm = new CalendarRecord(Alarm.Uri);
+				calAlarm.Set<CalendarTime>(Alarm.AlarmTime, time);
+				calAlarm.Set<int>(Alarm.TickUnit, (int)CalendarTypes.TickUnit.Specific);
+				calRecord.AddChildRecord(Event.Alarm, calAlarm);
+				calManager.Database.Update(calRecord);
+			}
+			catch (Exception ex)
+			{
+				throw new Exception(ex.Message);
+			}
+			finally
+			{
+				calAlarm?.Dispose();
+				calAlarm = null;
+				calRecord?.Dispose();
+				calRecord = null;
+				calManager?.Dispose();
+				calManager = null;
+			}
+
+			return true;
+		}
+
+		/// <summary>
+		/// Removes a calendar and all its events from the system.
+		/// </summary>
+		/// <param name="calendar">Calendar to delete</param>
+		/// <returns>True if successfully removed</returns>
+		/// <exception cref="System.ArgumentException">Calendar is read-only</exception>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<bool> DeleteCalendarAsync(Calendar calendar)
+		{
+			throw new PlatformNotSupportedException("Tizen platform doesn't support delete calendars");
+		}
+
+		/// <summary>
+		/// Removes an event from the specified calendar.
+		/// </summary>
+		/// <param name="calendar">Calendar to remove event from</param>
+		/// <param name="calendarEvent">Event to remove</param>
+		/// <returns>True if successfully removed</returns>
+		/// <exception cref="System.ArgumentException">Calendar is read-only</exception>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="System.InvalidOperationException">Editing recurring events is not supported</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public async Task<bool> DeleteEventAsync(Calendar calendar, CalendarEvent calendarEvent)
+		{
+			if (string.IsNullOrEmpty(calendar.ExternalID) || string.IsNullOrEmpty(calendarEvent.ExternalID))
+			{
+				return false;
+			}
+
+			var existingCal = await GetCalendarByIdAsync(calendar.ExternalID).ConfigureAwait(false);
+			if (existingCal == null)
+			{
+				return false;
+			}
+			else if (!existingCal.CanEditEvents)
+			{
+				throw new ArgumentException("Cannot delete event from readonly calendar", "calendar");
+			}
+			else
+			{
+				await Task.Run(() =>
+				{
+					CalendarManager calManager = null;
+					CalendarList calList = null;
+					try
+					{
+						calManager = new CalendarManager();
+						calList = calManager.Database.GetAll(Event.Uri, 0, 0);
+						calManager.Database.Delete(Event.Uri, Convert.ToInt32(calendarEvent.ExternalID));
+						calManager.Database.Delete(calList);
```

Comment:
Seems highly unexpected to delete _all_ calendar events when the caller only asked to delete _one_ event.